### PR TITLE
[MRG] Replace param `mask` with `mask_img` in First & Second Level Models for Nilearn parity

### PR DESCRIPTION
**Pull request by kchawla-pi**
Issue #315

### CODE REVIEW

File: nistats/tests/test_first_level_model.py [hunk 1]
```diff
@@ -445,3 +446,41 @@ def test_first_level_models_with_no_signal_scaling():
     assert_true(first_level_model.results_[0][0].theta.shape == (1, 1))
     # assert that the theta is equal to the one voxel value
     assert_almost_equal(first_level_model.results_[0][0].theta[0, 0], 6.0, 2)
+
+
+def test_params_deprecation_FirstLevelModel():
+    """ Tests whether use of deprecated keyword parameters of view_markers
+    raise corrrect warnings.
```
Comment (bthirion) [inline nistats/tests/test_first_level_model.py]: correct

**Comment by kchawla-pi** [comment]
Will make the deprecation at a few more places.

**Comment by bthirion** [comment]
Are you still working on it or can we merge ?

**Comment by kchawla-pi** [comment]
Yeah I am still working on it. The change has to be made for FirstLevelModel, SecondLevelModel, first_level_models_from_bids, and I am sure a couple more.  
I will change the status to MRG when it is ready for another review.

**Comment by kchawla-pi** [comment]
@bthirion lots of changes since the last review. 
`mask` parameter in `FirstLevelModel`, `SecondLevelModel`,`first_level_models_from_bids` has been replaced by `mask_img`, the way it is in Nilearn.
Any additional places which require the replacement?

**Comment by kchawla-pi** [comment]
Once the CI is green Review it. I consider this ready to merge.

**Comment by kchawla-pi** [comment]
It's the same code that I used earlier n this PR and in Nilearn. I only
added a couple lines of code to make them usable whenever we deprecate any
parameters in future.
So we don't have to write anything special again for that purpose ever
again, and we don't have to change the code or signature of any function.
I think it's totally worth the two extra lines.

**Comment by kchawla-pi** [comment]
We can discus what you think should be changed on Monday to make it lighter.

**Comment by bthirion** [comment]
Sorry I did not  check thoroughly. If the added code on the management of deprecation and argument handling is isomorphic to Nilearn, then we're fine. Will merge tomorrow unless I hear Otherwise.

**Pull request merged**